### ISSUE TITLE
Removing [locale] route at root

### ISSUE THREAD

**Issue by sshmaxime** [QuiiBz/next-international#188]
**Is your feature request related to a problem? Please describe.**
I'd like to get rid of the `[locale]` folder.

**Describe the solution you'd like**
Being able to define routes normally without the extra `[locale]` folder.

**Describe alternatives you've considered**
This library does it https://github.com/i18nexus/next-i18n-router but I haven't checked what's happening under the hood.

**Comment by gustaveWPM** [comment]
> That's something I want to get rid of too, thanks for opening a feature request. It's currently only used for translating in Client Components, more precisely inside `useCurrentLocale()` (which uses `useParams()`). We could theoretically retrieve the locale from the cookie instead of the route segment, but I haven't got the time yet to test this.

Hello,
What if we have some server component which use _locale_ from `params`?

**Comment by QuiiBz** [comment]
Indeed, this part will have to be updated to also retrieve the locale from the cookie, instead of the route params. I'm not aware of how to achieve this without opting-out of Static Rendering, so not really sure how this could be solved.

Does https://github.com/i18nexus/next-i18n-router supports Static Rendering? I couldn't find any documentation about it.

**Comment by radumkc** [comment]
[next-i18n-router](https://github.com/i18nexus/next-i18n-router) Does it by checking for a cookie that they set. If the cookie is there the locale can be determined in the middleware.

If the cookie is not there they use `@formatjs/intl-localematcher` and `negotiator` to autodetect and then they set the cookie.

It is fairly easy to understand I think, and the principle is simple to implement. The challenge is the detection as different websites have different settings.

**Comment by QuiiBz** [comment]
That's unfortunately not as easy. Using `cookie()` will opt-out of Static Rendering and force Dynamic Rendering instead (see my comment above: https://github.com/QuiiBz/next-international/issues/188#issuecomment-1730859664)

next-i18n-router also requires a dynamic segment (`[locale]`) to support Static Rendering:
<img width="1103" alt="Screenshot 2023-09-27 at 12 28 00" src="https://github.com/QuiiBz/next-international/assets/43268759/09fab8b2-360d-4269-ace9-40c0bd1ea17c">

**Comment by radumkc** [comment]
Yes of course, I should have been more specific. You can't do it for static unless you have a route I guess.

Not sure how it might work if you have domain level localisation instead of route based.

**Comment by gustaveWPM** [comment]
> Not sure how it might work if you have domain level localisation instead of route based.

You mean subdomains?
I don't know if it's changed since then, but it can be catastrophic in terms of SEO.

https://www.semrush.com/blog/subdomain-vs-subdirectory/

https://twitter.com/stekenwright/status/1318417365831385091?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1318417365831385091%7Ctwgr%5E1a7a1e7b2d7f8679ba057aa78ec887bfbfcce68d%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fwww.semrush.com%2Fblog%2Fsubdomain-vs-subdirectory%2F

https://twitter.com/digitalquokka/status/1321875238594949127?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1321875238594949127%7Ctwgr%5E1a7a1e7b2d7f8679ba057aa78ec887bfbfcce68d%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fwww.semrush.com%2Fblog%2Fsubdomain-vs-subdirectory%2F

(It remains to be seen if this impact is as dramatic for internationalisation purposes.)

**Comment by Bluzzi** [comment]
👋 Also interested in this feature! For a dashboard requiring a connection, it's not relevant to have a `[locale]` folder, glad this package takes this into consideration

**Comment by gustaveWPM** [comment]
> 👋 Also interested in this feature! For a dashboard requiring a connection, it's not relevant to have a `[locale]` folder, glad this package takes this into consideration

Maybe you don't need SSG in your app too...
Sooo, I _think_ you could omit the `[locale]` folder if you're using SSR everywhere, nope?

**Comment by Bluzzi** [comment]
> Maybe you don't need at all SSG in your app too... Sooo, I _think_ you could omit the `[locale]` folder if you're never using SSG in your app, nope?
> 
> I'm not sure about that... But I think the `[locale]` folder is mandatory only for SSG purposes?

I've just done a [small test](https://github.com/Bluzzi/NextJS-i18n) without the `[locale]` folder and it seems to work very well. 

But what exactly is the purpose of this ticket if it's already possible? To be able to do it with SSG support? Is it really possible without any information in the URL?

**Comment by gustaveWPM** [comment]
> > Maybe you don't need at all SSG in your app too... Sooo, I _think_ you could omit the `[locale]` folder if you're never using SSG in your app, nope?
> > I'm not sure about that... But I think the `[locale]` folder is mandatory only for SSG purposes?
> 
> I've just done a [small test](https://github.com/Bluzzi/NextJS-i18n) without the `[locale]` folder and it seems to work very well.
> 
> But what exactly is the purpose of this ticket if it's already possible? To be able to do it with SSG support? Is it really possible without any information in the URL?

Idk if it's possible. 😅
Glad to know that I could help a bit.
